tests/cupyx_tests/scipy_tests/ndimage_tests/test_filters.py

```diff
@@ -12,150 +11,335 @@
     pass
 
 
+class FilterTestCaseBase(unittest.TestCase):
+    """
+    Add some utility methods for the parameterized tests for filters. these
+    assume there are the "parameters" self.filter, self.wdtype or self.dtype,
+    and self.ndim, self.kshape, or self.shape. Other optional "parameters" are
+    also used if available like self.footprint when the filter is a filter
+    that uses the footprint. These methods allow testing across multiple
+    filter types much more easily.
+    """
+
+    # default param values if not provided
+    filter = 'convolve'
+    shape = (4, 5)
+    ksize = 3
+    dtype = numpy.float64
+    footprint = True
```

Comment:
I had previously suggested potentially testing inputs with order='F' to make sure everything is okay for inputs that don't have C order. To do that we could add an order parameter here that defaults to 'C' and then after `shaped_random` call:

```Python
if self.order == 'F':
    arr = xp.asfortranarray(arr)
```